src/Process/ProcessUtils.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace GrumPHP\Process;
+
+/**
+ * ProcessUtils is a bunch of utility methods.
+ *
+ * This class contains static methods only and is not meant to be instantiated.
+ *
+ * @author Martin Hasoň <martin.hason@gmail.com>
+ */
+class ProcessUtils
+{
+    /**
+     * Escapes a string to be used as a shell argument.
+     * Taken from the Symfony 4.0 package.
+     *
+     * @param string $argument The argument that will be escaped
+     *
+     * @return string The escaped argument
+     *
+     * @see \Symfony\Component\Process\Process::escapeArgument()
```

Comment:
please add a `TODO` that describes **when** and **how** get rid of this, as it's only for compatibility reasons

Reply:
It can be removed from the moment that Symfony's `Process` can accept an array of parameters (and thus, escape them itself again) (which seems to be [3.3](https://github.com/symfony/symfony/commit/330b61fecbef76a2e0c275cbfaee1dbeab176468)).

Little bit tricky answer, as that means from that moment on, we could / should then also remove the newly introduced-and-then-abstracted `ProcessArgumentsCollection::generateCliCommand()` method, too?  
Should I mark that method additionally as "deprecated" as of Symfony 3.3+?

Reply:
Put that knowledge into comment with `@todo` annotation so one will know it out of the box without searching for reasoning and solution when to drop it.

`ProcessArgumentsCollection::generateCliCommand()` class is public, if in best code approach (without working with legacy verion) we don't need this method, don't expose it to end user. so `@internal` it is (or even out of this class, as I would say it's weird to expose public classes with internal public methods, it's better to have public class and internal class).
`@deprecated` would be bad here, as deprecated code shall not be used even by the repo (except of testing those deprecations), and you call it obviously ;)

Reply:
This public method kinda has been [suggested by @veewee](https://github.com/phpro/grumphp/pull/428#discussion_r152907373).

Maybe I should go with his second approach instead?

> Maybe it's better to add a (...) `createProcess(): Process` method to the `ProcessArgumentsCollection` so that it is reusable?

But I initially went with the first, because the second was kinda hard-coupling the GrumPHP's `ProcessArgumentsCollection` to Symfony's `Process` component.

Reply:
You could also create an internal ProcessFactory that takes the ProcessArgumentsCollection as input? That way the logic is moved outside the class and can be changed to the array constructor in a later version?